studio/components/interfaces/Account/AddNewFactorModal.tsx

```diff
@@ -0,0 +1,201 @@
+import ConfirmationModal from 'components/ui/ConfirmationModal'
+import InformationBox from 'components/ui/InformationBox'
+import { GenericSkeletonLoader } from 'components/ui/ShimmeringLoader'
+import { useMfaChallengeAndVerifyMutation } from 'data/profile/mfa-challenge-and-verify-mutation'
+import { useMfaEnrollMutation } from 'data/profile/mfa-enroll-mutation'
+import { useMfaUnenrollMutation } from 'data/profile/mfa-unenroll-mutation'
+import { useStore } from 'hooks'
+import Image from 'next/image'
+import { Dispatch, SetStateAction, useEffect, useState } from 'react'
+import { Input, Modal } from 'ui'
+
+const AddNewFactorModal = ({ onClose }: { onClose: () => void }) => {
+  const [verificationComplete, setVerificationComplete] = useState(false)
+  // Generate a name with a number between 0 and 1000
+  const [name, setName] = useState(`App ${Math.floor(Math.random() * 1000)}`)
```

Comment:
This generates a `App 451` type of name. There was a mention of generating names using 2 nouns, do we have that type of generator anywhere else?

Reply:
we don't have, but i think that's okay